### PR TITLE
refactoring architecture

### PR DESCRIPTION
**Pull request by AlexeyAnurin**

### CODE REVIEW

File: app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import com.griddynamics.weatherappinternship.domain.repositoryInterfaces.WeatherRepository
+import com.griddynamics.weatherappinternship.data.WeatherRepositoryImpl
+import dagger.Binds
+import dagger.Module
+
+@Module
+abstract class RepositoryModule {
+
+    @Binds
+    internal abstract fun provideWetherRepository(weatherRepository: WeatherRepositoryImpl): WeatherRepository
```
Comment (sin-al-vl) [inline app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt:12]: This line is too long - more than 100 symbols

File: app/src/main/java/com/griddynamics/weatherappinternship/presentation/mainScreen/MainScreenViewModel.kt [hunk 2]
```diff
@@ -4,22 +4,25 @@ import android.util.Log
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.viewModelScope
 import com.griddynamics.weatherappinternship.BaseViewModel
-import com.griddynamics.weatherappinternship.data.WeatherRepository
+import com.griddynamics.weatherappinternship.domain.repositoryInterfaces.WeatherRepository
```
Comment (sin-al-vl) [inline app/src/main/java/com/griddynamics/weatherappinternship/presentation/mainScreen/MainScreenViewModel.kt:7]: I think `repositoryInterfaces` is too long. This package could have name just `repository`

File: app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt [hunk 3]
```diff
@@ -1,21 +1,18 @@
...
 import okhttp3.Interceptor
 import okhttp3.OkHttpClient
-import okhttp3.logging.HttpLoggingInterceptor
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
 
-@Module
+private const val BASE_URL = "https://api.openweathermap.org/data/2.5/"
+
+@Module(includes = arrayOf(InterceptorsModule::class))
```
Comment (sin-al-vl) [inline app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt:15]: Studio suggests than `arrayOf` should be replaced with array literal

File: app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt [hunk 4]
```diff
@@ -1,21 +1,18 @@
-package com.griddynamics.weatherappinternship.framework.di
+package com.griddynamics.weatherappinternship.di
```
Comment (sin-al-vl) [inline app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt:1]: 1. Move this back to `framowork.di`
2. Rename `framowork` to `network`
3. Move whole `netwrok` package to `data` package

File: app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.griddynamics.weatherappinternship.di
```
Comment (sin-al-vl) [inline app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt:1]: Please, move this to `data.repository.di`

File: app/src/main/java/com/griddynamics/weatherappinternship/data/WeatherRepositoryImpl.kt [hunk 6]
```diff
@@ -2,15 +2,16 @@ package com.griddynamics.weatherappinternship.data
```
Comment (sin-al-vl) [inline app/src/main/java/com/griddynamics/weatherappinternship/data/WeatherRepositoryImpl.kt:2]: Please, move this to `data.repository`

**Review by sin-al-vl — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt (hunk 1 above)
Comment (AlexeyAnurin) [reply app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt:12]: fixed

File: app/src/main/java/com/griddynamics/weatherappinternship/presentation/mainScreen/MainScreenViewModel.kt (hunk 2 above)
Comment (AlexeyAnurin) [reply app/src/main/java/com/griddynamics/weatherappinternship/presentation/mainScreen/MainScreenViewModel.kt:7]: fixed

File: app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt (hunk 3 above)
Comment (AlexeyAnurin) [reply app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt:15]: fixed

File: app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt (hunk 4 above)
Comment (AlexeyAnurin) [reply app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt:1]: fixed

File: app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt (hunk 5 above)
Comment (AlexeyAnurin) [reply app/src/main/java/com/griddynamics/weatherappinternship/di/RepositoryModule.kt]: fixed

File: app/src/main/java/com/griddynamics/weatherappinternship/data/WeatherRepositoryImpl.kt (hunk 6 above)
Comment (AlexeyAnurin) [reply app/src/main/java/com/griddynamics/weatherappinternship/data/WeatherRepositoryImpl.kt]: fixed

File: app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt (hunk 4 above)
Comment (sin-al-vl) [reply app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt]: This comment was only about `NetworkModule`. You shouldn't did all those steps for everything in root `di` package. Please, revert everything except `NetworkModule` back to root `di` package (scope, AppComponent & AppModule).

Comment (AlexeyAnurin) [reply app/src/main/java/com/griddynamics/weatherappinternship/di/NetworkModule.kt]: fixed

**Review by sin-al-vl — APPROVED** [review]

**Pull request merged**